Xamarin.PropertyEditing.Tests/RatioViewModelTests.cs

```diff
@@ -0,0 +1,182 @@
+		}
+
+		[Test]
+		public void RatioSeparator ()
+		{
+			var property = GetPropertyMock ();
+			var editor = GetBasicEditor (property.Object);
+			var vm = GetViewModel (property.Object, new[] { editor });
+			Assume.That (vm.Value, Is.EqualTo (new CommonRatio (3, 1, ':')));
+
+			bool xChanged = false, valueChanged = false;
+			vm.PropertyChanged += (sender, args) => {
+				if (args.PropertyName == nameof (RatioViewModel.RatioSeparator))
+					xChanged = true;
+				if (args.PropertyName == nameof (RatioViewModel.Value))
+					valueChanged = true;
+			};
+
+			vm.RatioSeparator = '/';
+			Assert.That (vm.Value.RatioSeparator, Is.EqualTo ('/'));
+			Assert.That (xChanged, Is.True);
+			Assert.That (valueChanged, Is.True);
+		}
+
+		[Test]
+		public void ValueChangesNumeratorDenominatorRatioSeparator ()
+		{
+			var property = GetPropertyMock ();
+			var editor = GetBasicEditor (property.Object);
+			var vm = GetViewModel (property.Object, new[] { editor });
+			Assume.That (vm.Numerator, Is.EqualTo (0));
+			Assume.That (vm.Denominator, Is.EqualTo (0));
+
+			bool nChanged = false, dChanged = false, sChanged = false, valueChanged = false;
+			vm.PropertyChanged += (sender, args) => {
+				if (args.PropertyName == nameof(RatioViewModel.Numerator))
+					nChanged = true;
+				if (args.PropertyName == nameof(RatioViewModel.Denominator))
+					dChanged = true;
+				if (args.PropertyName == nameof (RatioViewModel.RatioSeparator))
+					sChanged = true;
+				if (args.PropertyName == nameof(SizePropertyViewModel.Value))
+					valueChanged = true;
+			};
+
+			vm.Value = new CommonRatio (5, 10, ':');
+
+			Assert.That (vm.Numerator, Is.EqualTo (5));
+			Assert.That (vm.Denominator, Is.EqualTo (10));
+			Assert.That (nChanged, Is.True);
+			Assert.That (dChanged, Is.True);
+			Assert.That (sChanged, Is.True);
+			Assert.That (valueChanged, Is.True);
+		}
+
+		[Test]
+		public void RatioWithoutDenominator ()
+		{
+			var property = GetPropertyMock ();
+			var editor = GetBasicEditor (property.Object);
+			var vm = GetViewModel (property.Object, new[] { editor });
+
+			vm.SetRatioFromString ("21");
+
+			Assert.That (vm.Numerator, Is.EqualTo (21));
+			Assert.That (vm.Denominator, Is.EqualTo (1));
+			Assert.That (vm.RatioSeparator, Is.EqualTo (':'));
+		}
+
+		[Test]
+		public void RatioWithNumeratorDenominatorAndColonSeparator ()
+		{
+			var property = GetPropertyMock ();
+			var editor = GetBasicEditor (property.Object);
+			var vm = GetViewModel (property.Object, new[] { editor });
+
+			vm.SetRatioFromString ("21:2");
+
+			Assert.That (vm.Numerator, Is.EqualTo (21));
+			Assert.That (vm.Denominator, Is.EqualTo (2));
+			Assert.That (vm.RatioSeparator, Is.EqualTo (':'));
+		}
+
+		[Test]
+		public void RatioWithNumeratorDenominatorAndSlashSeparator ()
+		{
+			var property = GetPropertyMock ();
+			var editor = GetBasicEditor (property.Object);
+			var vm = GetViewModel (property.Object, new[] { editor });
+
+			vm.SetRatioFromString ("21/2");
```

Comment:
I'd add a test case with spaces, to finish addressing my above comment.

Reply:
Yes, good idea.